discovery_engine/lib/src/domain/feed_manager.dart

```diff
@@ -38,16 +40,31 @@ class FeedManager {
   /// Handle the given feed client event.
   ///
   /// Fails if [event] does not have a handler implemented.
-  Future<void> handleFeedClientEvent(FeedClientEvent event) async {
-    await event.maybeWhen(
+  /// If [event] is [FeedRequested], returns the feed documents.
+  /// Returns an empty list in all other cases.
+  Future<List<Document>> handleFeedClientEvent(FeedClientEvent event) {
```

Comment:
I guess I misinterpreted you message when we talked about this.
Will be difficult for the handler to then return the correct message with this type. 
`feedRequested` return a list of documents like `nextFeedBatchRequested` but `feedDocumentsClosed` return only `ClientRequestSuccesful`.
@biernacki do you have any good idea on how to handle this? I guess one option will be to have two different methods or return a more descriptive type or return the event itself.

Reply:
Well, at this time we should just use `EngineEvent` as the return type from the `FeedManager` cause it contains all the possible responses (when we add later more events, like search related ones, that will change).

We could also think about a different structure for Engine events where we have for each client event freezed union with possible responses. But maybe we could revisit that option later. We could combine it then with another option, which would be to have usecases instead of managers, one usecase per event.

Reply:
thanks :+1: will try `EngineEvent` as the return type for now then and see how that goes. btw I notice it doesn't have a response for `feedDocumentsClosed` so will add a `FeedDocumentsClosedSuccess` (or similar)..

Reply:
the intended response for `FeedDocumentsClosed` is atm `ClientEventSucceeded`, but ofc. we can change that. Though I feel we should be consistent and maybe add success/failure response events to all client events (maybe with freezed union classes)

Reply:
thanks, since `ClientEventSucceeded` is the intended return value I've gone for that.